In [1]:
import requests
import getpass

1. First try for World Trading Data API to get stock market values

In [ ]:
api_token = getpass.getpass()
url = "https://api.worldtradingdata.com/api/v1/history?symbol=SNAP&date=2019-01-02&api_token=" + api_token

resp = requests.get(url)

In [3]:
resp.json()

{'name': 'SNAP',
 'history': {'2020-04-03': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-04-02': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-04-01': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-31': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-30': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-27': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-26': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-25': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-24': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-23': {'open': None,
   'close': None,
   'h

In [4]:
url = "https://api.worldtradingdata.com/api/v1/history_multi_single_day?symbol=DAX,TWTR&date=2019-01-02&api_token=" + api_token

resp = requests.get(url)
resp.json()

{'date': '2019-01-02',
 'data': {'TWTR': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None}}}

2. We got none results for history and multi single day data beacuse a premium account is needed, we tried different requests with different stocks and got none results

In [5]:
url = "https://api.worldtradingdata.com/api/v1/history?symbol=DJI,TWTR&date=2019-01-02&api_token="  + api_token

resp = requests.get(url)
resp.json()

{'Message': 'Error! The requested stock could not be found.'}

In [6]:
url = "https://api.worldtradingdata.com/api/v1/stock?symbol=DJI&api_token="  + api_token

resp = requests.get(url)
resp.json()

{'Message': 'Error! The requested stock(s) could not be found.'}

In [7]:
url = "https://api.worldtradingdata.com/api/v1/intraday?symbol=SNAP&api_token=" + api_token

resp = requests.get(url)
resp

<Response [404]>

In [8]:
url = "https://api.worldtradingdata.com/api/v1/stock?symbol=SNAP,TWTR&api_token=" + api_token

resp = requests.get(url)

resp.json()

{'symbols_requested': 2,
 'symbols_returned': 2,
 'data': [{'symbol': 'SNAP',
   'name': 'Snap Inc.',
   'currency': 'USD',
   'price': None,
   'price_open': None,
   'day_high': None,
   'day_low': None,
   '52_week_high': None,
   '52_week_low': None,
   'day_change': None,
   'change_pct': None,
   'close_yesterday': None,
   'market_cap': None,
   'volume': None,
   'volume_avg': None,
   'shares': None,
   'stock_exchange_long': 'New York Stock Exchange',
   'stock_exchange_short': 'NYSE',
   'timezone': 'EDT',
   'timezone_name': 'America/New_York',
   'gmt_offset': '-14400',
   'last_trade_time': 'N/A',
   'pe': None,
   'eps': None},
  {'symbol': 'TWTR',
   'name': 'Twitter, Inc.',
   'currency': 'USD',
   'price': None,
   'price_open': None,
   'day_high': None,
   'day_low': None,
   '52_week_high': None,
   '52_week_low': None,
   'day_change': None,
   'change_pct': None,
   'close_yesterday': None,
   'market_cap': None,
   'volume': None,
   'volume_avg': None,
   'shar

In [9]:
url = "https://intraday.worldtradingdata.com/api/v1/intraday?symbol=SNAP&api_token=YAbbxHWrJip30D8HGEdhGXaIYxQ2MJnkprP56aY67iQ1D7MvLCJJbMdPDVgT"
resp = requests.get(url)

resp.json()

{'symbol': 'SNAP',
 'stock_exchange_short': 'NYSE',
 'timezone_name': 'America/New_York',
 'intraday': {'2020-04-03 09:30:00': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None}}}

In [11]:
url = "https://api.worldtradingdata.com/api/v1/history?symbol=TWTR&api_token=" + api_token

resp = requests.get(url)

resp.json()

{'name': 'TWTR',
 'history': {'2020-04-03': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-04-02': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-04-01': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-31': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-30': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-27': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-26': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-25': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-24': {'open': None,
   'close': None,
   'high': None,
   'low': None,
   'volume': None},
  '2020-03-23': {'open': None,
   'close': None,
   'h

3. Trying second API, IEXPAIS, for reading stock market values

In [ ]:
token = getpass.getpass()

In [12]:
url = "https://cloud.iexapis.com/stable/stock/dax/ohlc?token=" + token

resp = requests.get(url)

resp

<Response [402]>

Error 402 means that we don't have access with our credentials